src/curve/LinearIncreasingCurve.sol

```diff
@@ -424,7 +424,7 @@ contract LinearIncreasingCurve is
                 _fromLocked.amount
             );
 
-            if (_newLocked.amount == 0) {
+            if (_newLocked.amount < _fromLocked.amount) {
                 lastPoint.bias -= oldLockBias;
```

Comment:
I'm not sure this is right. I think the correct generalisation is:

- Remove the old slope and bias from the lastPoint
- If the lock is not mature, remove it from the dslopes
- Then add the newLock bias slope and slope cchangee